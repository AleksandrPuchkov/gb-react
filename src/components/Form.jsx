import React, { useState } from 'react'
import './styles/Form.css'
import { Parent } from './Parent'

export const Form = () => {

    return <div>
       <Parent />
    </div>
}