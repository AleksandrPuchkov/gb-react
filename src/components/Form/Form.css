form {
    background-color: #DDD;
    padding         : 15px;
    margin          : 0 auto;
    width           : 400px;
}

.inputusername, .inputmessage {
    padding: 5px;
    height: 20px;
    border-radius: 5px;
    margin-bottom: 10px;
}

.inputusername {

}

.inputmessage {
    width: 390px;
    height: 100px;

}

.submitbutton {
    padding: 5px;
    height: 30px;
}

.submitbutton:hover {cursor: pointer}