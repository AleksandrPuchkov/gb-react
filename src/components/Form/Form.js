"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Form = void 0;
const react_1 = __importStar(require("react"));
const nanoid_1 = require("nanoid");
const MessageList_1 = require("../MessageList/MessageList");
const Button_1 = require("../Button/Button");
require("./Form.css");
const Form = () => {
    const [message, setValue] = (0, react_1.useState)('');
    const [username, setUsername] = (0, react_1.useState)('');
    const [messages, setMessages] = (0, react_1.useState)([]);
    const [disabled, setDisabled] = (0, react_1.useState)(false);
    const handleSubmitForm = (e) => {
        e.preventDefault();
        setMessages([...messages, { id: (0, nanoid_1.nanoid)(), username: username, message: message }]);
        setValue('');
    };
    (0, react_1.useEffect)(() => {
        if (messages.length &&
            messages[messages.length - 1].username != 'Chatbot') {
            setDisabled(!disabled);
            const botTimeout = setTimeout(() => {
                setMessages([
                    ...messages,
                    { id: (0, nanoid_1.nanoid)(), username: 'Chatbot', message: 'Hello from Chatbot!' },
                ]);
            }, 1500);
            return () => {
                clearTimeout(botTimeout);
            };
        }
        if (disabled)
            setDisabled(!disabled);
    }, [messages]);
    return (react_1.default.createElement("form", { onSubmit: handleSubmitForm },
        react_1.default.createElement(MessageList_1.MessageList, { messages: messages }),
        react_1.default.createElement("input", { className: "inputusername", "data-testid": "inputusername", placeholder: "\u0412\u0432\u0435\u0434\u0438\u0442\u0435 \u0438\u043C\u044F...", type: "text", value: username, onChange: (e) => setUsername(e.target.value) }),
        react_1.default.createElement("br", null),
        react_1.default.createElement("br", null),
        react_1.default.createElement("textarea", { className: "inputmessage", placeholder: "\u0412\u0432\u0435\u0434\u0438\u0442\u0435 \u0442\u0435\u043A\u0441\u0442...", value: message, onChange: (e) => setValue(e.target.value) }),
        react_1.default.createElement("br", null),
        react_1.default.createElement(Button_1.Button, { disabled: disabled })));
};
exports.Form = Form;
