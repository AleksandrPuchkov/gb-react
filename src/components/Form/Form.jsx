import React, { useState, useEffect } from 'react';
import { MessageList } from '../MessageList/MessageList';
import { Button } from '../Button/Button';
import './Form.css';

export const Form = () => {
  const [message, setValue] = useState('');
  const [username, setUsername] = useState('');
  const [messages, setMessages] = useState([]);
  const [disabled, setDisabled] = useState(false);

  const handleSubmitForm = (e) => {
    e.preventDefault();
    setMessages([...messages, { username: username, message: message }]);
    setValue('');
  };

  useEffect(() => {
    if (
      messages.length &&
      messages[messages.length - 1].username != 'Chatbot'
    ) {
      setDisabled(!disabled);
      const botTimeout = setTimeout(() => {
        setMessages([
          ...messages,
          { username: 'Chatbot', message: 'Hello from Chatbot!' },
        ]);
      }, 1500);

      return () => {
        clearTimeout(botTimeout);
      };
    }
    if (disabled) setDisabled(!disabled);
  }, [messages]);

  return (
    <form onSubmit={handleSubmitForm}>
      <MessageList messages={messages} />
      <input
        className="inputusername"
        data-testid="inputusername"
        placeholder="Введите имя..."
        type="text"
        value={username}
        onChange={(e) => setUsername(e.target.value)}
      />
      <br />
      <textarea
        className="inputmessage"
        placeholder="Введите текст..."
        type="text"
        value={message}
        onChange={(e) => setValue(e.target.value)}
      />
      <br />
      <Button disabled={disabled} />
    </form>
  );
};
