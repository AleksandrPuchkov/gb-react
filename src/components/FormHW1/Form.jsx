import React, { useState } from 'react'
import './styles/Form.css'
import { Message } from './Message'
import { Input } from './Input'
import { Button } from './Button'

export const Form = () => {
    const [value, setValue] = useState('')
    const [message, setMessage] = useState('Это поле Message, и в нем отобразится новое сообщение')

    const handleClick = () => {
        setMessage(value)
    }

    const handleChange = (ev) => {
        setValue(ev.target.value)
    }

    return <div className='form'>
        <Message message={message} />
        <Input value={value} change={handleChange}  />
        <br />
        <Button click={handleClick} />
    </div>
}