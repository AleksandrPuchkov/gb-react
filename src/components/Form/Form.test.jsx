import React from 'react';
import { Form } from './Form';
import { fireEvent, render, screen } from '@testing-library/react';

describe('Form', () => {
  it('Render component', () => {
    render(<Form />);
  });

  it('render with snapshot', () => {
    const { asFragment } = render(<Form />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Entering username', () => {
    render(<Form />);
    fireEvent.change(screen.getByPlaceholderText('Введите имя...'), {
      target: { value: 'TestUsername' },
    });
    expect(screen.getByPlaceholderText('Введите имя...').value).toBe(
      'TestUsername'
    );
  });

  it('Remember username', () => {
    render(<Form />);
    fireEvent.change(screen.getByPlaceholderText('Введите имя...'), {
      target: { value: 'TestUsername' },
    });
    fireEvent.click(screen.getByTestId('submitbutton'));
    expect(screen.getByPlaceholderText('Введите имя...').value).toBe(
      'TestUsername'
    );
  });

  it('Entering message', () => {
    render(<Form />);
    fireEvent.input(screen.getByPlaceholderText('Введите текст...'), {
      target: { value: 'TestMessage' },
    });
    expect(screen.getByPlaceholderText('Введите текст...').innerHTML).toBe(
      'TestMessage'
    );
  });

  it('Cleansing message field', () => {
    render(<Form />);
    fireEvent.input(screen.getByPlaceholderText('Введите текст...'), {
      target: { value: 'TestMessage' },
    });
    fireEvent.click(screen.getByTestId('submitbutton'));
    expect(screen.getByPlaceholderText('Введите текст...').innerHTML).toBe('');
  });

  it('Displaying user`s message', () => {
    render(<Form />);
    fireEvent.change(screen.getByPlaceholderText('Введите имя...'), {
      target: { value: 'TestUsername' },
    });
    fireEvent.input(screen.getByPlaceholderText('Введите текст...'), {
      target: { value: 'TestMessage' },
    });
    fireEvent.click(screen.getByTestId('submitbutton'));
    expect(screen.getByRole('listitem').firstChild.innerHTML).toBe(
      'TestUsername'
    );
    expect(screen.getByRole('listitem').lastChild.innerHTML).toBe(
      'TestMessage'
    );
  });

  it('Bot`s response', async () => {
    render(<Form />);
    fireEvent.input(screen.getByPlaceholderText('Введите текст...'), {
      target: { value: 'TestMessage' },
    });
    fireEvent.click(screen.getByTestId('submitbutton'));
    expect(await screen.findByText('Chatbot', {}, { timeout: 2000 })).toBeVisible;
  });
});
