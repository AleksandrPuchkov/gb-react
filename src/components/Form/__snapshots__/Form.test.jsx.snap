// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Foasdasdrm render with snapshot 1`] = `
<DocumentFragment>
  <form>
    <ul />
    <input
      class="inputusername"
      data-testid="inputusername"
      placeholder="Введите имя..."
      type="text"
      value=""
    />
    <br />
    <textarea
      class="inputmessage"
      placeholder="Введите текст..."
      type="text"
    />
    <br />
    <button
      class="submitbutton"
      data-testid="submitbutton"
      type="submit"
    >
      Отправить
    </button>
  </form>
</DocumentFragment>
`;

exports[`Form render with snapshot 1`] = `
<DocumentFragment>
  <form>
    <ul />
    <input
      class="inputusername"
      data-testid="inputusername"
      placeholder="Введите имя..."
      type="text"
      value=""
    />
    <br />
    <textarea
      class="inputmessage"
      placeholder="Введите текст..."
      type="text"
    />
    <br />
    <button
      class="submitbutton"
      data-testid="submitbutton"
      type="submit"
    >
      Отправить
    </button>
  </form>
</DocumentFragment>
`;
